### DISCUSSION TITLE
How can I obtain the reaction rate of species k in PeleC?

### DISCUSSION THREAD

**Discussion by Rachel008996** [AMReX-Combustion/PeleC#773] [Q&A]
In the data I have, there are outputs similar to ('boxlib', 'rho_omega_CO'), which I believe should be the generation rate of CO.

**Comment by drummerdoc** [comment]
Please rephrase the question.  D(species) is the species transport coefficient, which can be spatially and temporally varying, due to its optional dependence on temperature and composition.  rho_omega_CO would be the net mass production rate of CO, computed as a snapshot given the state at the time of evaluation.  For PeleC, this would be in grams of CO per cm^3 per second.  Note that if you want to use this information for reaction flux or displacement speed analysis there are subtleties about how and where this is evaluated.  As for "cgs causing confusion" I'm not sure how to help.

**Reply by drummerdoc** [reply]
Also, in both cases adding this to the plot files would generate the information based on the cell-centered state at the time the plot files are written.  Transport coefficients get interpolated to cell faces prior to use, and species production rates get evaluated within a subcyled time integrator.  So, reconstructed discrete balanced using these things is pretty tricky, especially looking at the transport fluxes near coarse fine interfaces. If you are trying to recover precisely how the conservation equation are integrated in time, it will require digging deep in the code.  So, please be aware of how you plan to use those quantities.  Ask for help, if needed.
